test/parallel/test-promises-unhandled-rejections.js

```diff
@@ -702,19 +703,22 @@ asyncTest('Rejected promise inside unhandledRejection allows nextTick loop' +
 });
 
 asyncTest(
-  'Unhandled promise rejection emits a warning immediately',
+  'Promise rejection triggers unhandledRejection immediately',
   function(done) {
     clean();
-    Promise.reject(0);
-    const { emitWarning } = process;
-    process.emitWarning = common.mustCall((...args) => {
+    process.on('unhandledRejection', common.mustCall((err) => {
       if (timer) {
         clearTimeout(timer);
         timer = null;
         done();
       }
-      emitWarning(...args);
-    }, 2);
+    }));
+    try {
+      Promise.reject(0);
+      assert(false);
+    } catch (e) {
+      assert(!!e);
+    }
```

Comment:
I am not certain what these lines are meant to verify. Unhandled rejections can't be detected sync. Thus, it should also be fine to keep the `Promise.reject(0)` at the top where it was before.